### PR TITLE
[JEP-305] Incrementalify

### PR DESCRIPTION
**Pull request by batmat**
See https://github.com/jenkinsci/incrementals-tools

This would be cool to be able to generate _Incrementals_ release on each `master` merge, or for PRs, like it should happen for this one already.

* RequireUpperBoundDeps fix induced by the parent pom bump
* Bump to Java 8 / Jenkins 2.60.3 as minimum version
  (I had a local test failure on older core, and I do not think it still
  makes sense to spend time even trying to understand why it fails on
  Java 7 and Jenkins 1.642.4).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1 +1 @@
-buildPlugin(jenkinsVersions: [null, '2.32.3'])
```
Comment (batmat) [inline Jenkinsfile]: cc @jglick esp. for this change, given you were the one adding it.

File: pom.xml [hunk 2]
```diff
@@ -79,6 +82,12 @@
             <artifactId>structs</artifactId>
             <version>1.5</version>
             <scope>test</scope>
+            <exclusions> <!-- coming from the core - RequireUpperBoundDeps fix -->
+                <exclusion>
+                    <groupId>org.jenkins-ci</groupId>
+                    <artifactId>annotation-indexer</artifactId>
```
Comment (dwnusbaum) [inline pom.xml]: nit: I think you could update to structs 1.10+ to pick up https://github.com/jenkinsci/structs-plugin/commit/61baf803b832dce4fe5382ada874625a9a7bd6fc#diff-4773b53940bf034d91df6ea6b029028e which moves that dependency into provided scope instead, although that might necessitate other RequireUpperBoundDeps fixes.

File: Jenkinsfile (hunk 1 above)
Comment (jglick) [reply Jenkinsfile]: fine

File: pom.xml (hunk 2 above)
Comment (jglick) [reply pom.xml]: Yup, just update `structs`.

**Comment by batmat** [comment]
Tests failing on Jenkins 2.32.3, because my `Jenkinsfile` change didn't get used because I'm not a committer here.
image

**Pull request merged**